### ISSUE TITLE
Thread Killed, Invalid Request, Request Body size not complete

### ISSUE THREAD

**Issue by YipYup** [astrada/google-drive-ocamlfuse#301]
## Behavior Observed
Got what appears to be a non-fatal exception when running in foreground mode.

`Thread 532871 killed on uncaught exception Failure("Error: Invalid request.  There were 1056899072 byte(s) in the request body.  There should have been 1388505728 byte(s) (starting at offset 0 and ending at offset 1388505727) according to the Content-Range header. (HTTP response code: 400)")`

## Version
ppa:alessandro-strada/google-drive-ocamlfuse-beta
```
root@localhost:/# google-drive-ocamlfuse -version
google-drive-ocamlfuse, version 0.6.19
Copyright (C) 2012-2017 Alessandro Strada
License MIT
```

## Environment

Launched as:

`root@localhost:~# google-drive-ocamlfuse -label email@domain.com /mnt/gdrive -f -m -o rw,atomic_o_trunc,allow_other,dev,suid`

Config File:
```
apps_script_format=json
apps_script_icon=
async_upload=true
cache_directory=
client_id= [ REDACTED ]
client_secret= [ REDACTED ]
connect_timeout_ms=5000
curl_debug_off=false
data_directory=
delete_forever_in_trash_folder=false
docs_file_extension=true
document_format=odt
document_icon=
download_docs=true
drawing_format=png
drawing_icon=
form_format=zip
form_icon=
fusion_table_format=desktop
fusion_table_icon=
keep_duplicates=false
large_file_read_only=false
large_file_threshold_mb=1
log_directory=/var/log/google-drive-ocamlfuse/
lost_and_found=false
low_speed_limit=0
low_speed_time=0
map_format=desktop
map_icon=
max_cache_size_mb=2048
max_download_speed=0
max_memory_cache_size=1073741824
max_retries=8
max_upload_chunk_size=1099511627776
max_upload_speed=0
memory_buffer_size=10485760
metadata_cache_time=60
presentation_format=pdf
presentation_icon=
read_ahead_buffers=3
read_only=false
shared_with_me=false
spreadsheet_format=ods
spreadsheet_icon=
sqlite3_busy_timeout=5000
stream_large_files=true
umask=0o000
verification_code=
```

**Comment by YipYup** [comment]
One more example.

`Thread 680595 killed on uncaught exception Failure("Error: Invalid request.  There were 820510720 byte(s) in the request body.  There should have been 1415072573 byte(s) (starting at offset 0 and ending at offset 1415072572) according to the Content-Range header. (HTTP response code: 400)")`

**Comment by YipYup** [comment]
`Thread 1024388 killed on uncaught exception Failure("Error: Invalid request.  There were 996212736 byte(s) in the request body.  There should have been 1491472403 byte(s) (starting at offset 0 and ending at offset 1491472402) according to the Content-Range header. (HTTP response code: 400)")`

**Comment by YipYup** [comment]
From what I can tell, each time this happens, the file transfer from cache to Google Drive does not finish properly, and I end up with a 0-byte file on my Google Drive, to say that whatever file I am moving or copying, ends up bad on Google Drive. I suppose this is acceptable if you're doing a cp instead of a mv, but It has resulted in losing my data when using mv.

**Comment by YipYup** [comment]
Thread 1856853 killed on uncaught exception Failure("Error: Invalid request.  There were 630784000 byte(s) in the request body.  There should have been 991414107 byte(s) (starting at offset 0 and ending at offset 991414106) according to the Content-Range header. (HTTP response code: 400)")
Segmentation fault

**Comment by astrada** [comment]
Should be fixed in 0.6.20 (beta channel).
